gokey/cache.go

```diff
@@ -10,69 +10,65 @@ type Cache struct {
 }
 
 type pair struct {
-	ttl   time.Duration
-	value []byte
+	ttl       time.Duration
+	createdAt time.Time
+	value     []byte
 }
 
 var (
 	_ Operations = (*Cache)(nil)
 
-	ErrorEmptyKey = errors.New("key cannot be empty")
+	ErrEmptyKey       = errors.New("key cannot be empty")
+	ErrExpiredKey     = errors.New("key expired")
+	ErrNoExistKey     = errors.New("key does not exist")
+	ErrNotImplemented = errors.New("not implemented")
 )
 
 // Get the values of the key, if this exists in the cache
-func (this *Cache) Get(key string) ([]byte, error) {
+func (c *Cache) Get(key string) ([]byte, error) {
 	if isEmpty(key) {
-		return nil, ErrorEmptyKey
+		return nil, ErrEmptyKey
 	}
 
 	keyEncrypted := generateMD5HashFromKey([]byte(key))
-	pair, exists := this.pairsSet[keyEncrypted]
+	pair, exists := c.pairsSet[keyEncrypted]
+	if !exists {
+		return nil, ErrNoExistKey
+	}
 
-	if exists {
-		return pair.value, nil
+	if time.Since(pair.createdAt) > pair.ttl && pair.ttl != -1 {
+		delete(c.pairsSet, keyEncrypted)
```

Comment:
@eze8789 no me convence el delete este aca. Creo que deberia haber un mecanismo diferente / aparte para expirar las keys.
Igualmente, por mas que expiren yo las trataria como not found en lugar de tener 2 resultados diferentes con la misma pegada 2 veces. Que opinas?

Reply:
Respecto al delete básicamente tome lo que redis llama el lazy expiration para hacer un clean up en caso de que el proceso de limpieza no haya borrado una key y tengamos un get a un elemento expirado. Lo dejo a decisión lo de removerlo, coincido que tenemos que contar con algun otro proceso que esté también haciendo el clean up por una cuestión de limpiar de memoria cosas que ya no sirven y no se acceden, esto es solo un layer de clean up extra. 
En cuanto al mensaje de error me parece ok y estoy de acuerdo que no agrega valor tener logs messages distintos, lo cambio en el próximo commit según que se prefiera hacer con el delete.

Reply:
Tengamos un solo error de not found, cuando la key no esta en el cache, ya sea por que realmente es un cache miss o porque expiró.
En cuanto al delete lo veo bien por ahora, habria que investigar otras maneras de hacerlo durante el runtime, aunque puede volverse mas complejo para esta etapa.

Reply:
Normalizado el error message